src/cache/cache_las.c

```diff
@@ -1516,6 +1597,12 @@ __wt_find_lookaside_upd(
 
 err:	__wt_readunlock(session, &cache->las_sweepwalk_lock);
 	WT_TRET(__wt_las_cursor_close(session, &cursor, session_flags));
+	if (listp == NULL)
```

Comment:
Are we leaking memory when ```listp``` is not null, in that case we do not free ```list``` that we copied to ```listp```. Can that be potentially an issue with the original code in ```__wt_value_return_upd ``` as well?

Reply:
That null check is for if we jump to `err` due to a failed `__wt_realloc_def` in which case `listp` will be `NULL` and won't have a heap allocated list assigned to it. So I just assign it back to `list` and free the updates in `list`.

I check `allocated_bytes` to see whether I exceeded the capacity of `list` and had to do the realloc. In that case, I free the underlying list also.

Reply:
You are right, everything seems accounted for.